### ISSUE TITLE
Workflow execution with a custom function fails with "Workflow currently unavailable"

### ISSUE THREAD

**Issue by BSouzaDock** [slack-samples/bolt-python-custom-function-template#7]
Hello,

I'm testing the custom functions with slack bolt, I followed the entire tutorial. I run the app and see "Bolt app is running!" in my terminal.

I can see the custom step that I created and I can link it to my wokflow and save it successfully.

The problem is when I try to run the workflow, the following error appears:

![image](https://github.com/slack-samples/bolt-python-custom-function-template/assets/96547309/5bb6c6c7-bc03-4526-877f-90d61717c2e4)

Can you help me with this? Thanks.

**Comment by zimeg** [comment]
Hey @BSouzaDock 👋  Sorry to hear you're getting caught by this error! I think this might be related to some workspace permissions instead of this sample, but want to get to the bottom of this in any case.

Are you trying to run your workflow in a developer sandbox or a production workspace? This shouldn't make a difference but it might be useful in checking a few things 🤔  Another thing to possibly check is if the "Default visibility for third-party steps" setting is on from the Admin settings: **Settings > Organization policies**.

I'm also curious if any logs appear for your app or if this error is appearing immediately when you try to start the workflow.

**Comment by BSouzaDock** [comment]
Hello @zimeg ,

Thanks for the answer.

We don't have a Sandox environment here, I'm doing it in production.
I didn't find this option "Default visibility for third-party steps" in Settings & Permissions.

I went to the workflow activity logs and no logs appear. The error appears every time the workflow is executed.

The configuration looked like this:
![image](https://github.com/slack-samples/bolt-python-custom-function-template/assets/96547309/66b6e2c5-9f97-4261-8899-03e292d378ce)

**Comment by cchensh** [comment]
Hi @BSouzaDock , is it possible for you to share your workspace ID?

**Comment by BSouzaDock** [comment]
Hello @cchensh ,

Wouldn't this present a security risk? I can't send you a private message.

**Comment by cchensh** [comment]
Hi @BSouzaDock , it makes sense, maybe you can send it to @zimeg 's email if you are okay with that. If not, based on the issue you shared here, I'm wondering is your workspace in a free plan or it's in a paid plan. If it's a free plan, you won't be able to run a workflow that contains custom functions, if you are in a paid plan, it worth checking with your workspace admin/owner to see if you still have premium workflow executions left.

**Comment by BSouzaDock** [comment]
@cchensh , In this case, I am the workspace admin. Here we have the enterprise select plan.
I've been developing using bolt for some time now and we have more than 10 woksflow steps from the legacy version.
I'm considering converting to the new version, as many people use it, and if it stops working in September, it will cause a big problem.

**Comment by BSouzaDock** [comment]
I checked the organization's workflow list. Only the one I created is marked as premium. But as there is no other, I believe that the error is not due to this, since there is no existing premium workflow execution, the execution of this would not generate expenses.

**Comment by zimeg** [comment]
@BSouzaDock still not certain what might be causing this, but can you check that the "Complimentary" workflow run count is in range here: https://app.slack.com/manage/E0123456789/integrations/workflows (replaced with your enterprise ID)

**Comment by BSouzaDock** [comment]
Hi @zimeg ,

I do not have access to this enterprise id. It seems that this enterprise select license provides some features of the enterprise grid license, but does not create an organization.
I have access to this screen. The only premium wokflow is the one I created: ![image](https://github.com/slack-samples/bolt-python-custom-function-template/assets/96547309/abc77004-9b74-4b1e-85e8-ace1bcecc49f)

I created a sandbox account and ran tests on the same project and it worked normally. I have an open ticket on slack and they are checking it.
